### PR TITLE
✨The label cluster-name is now applied to external objects

### LINKED ISSUE #1567: Add cluster label to infrastructure and bootstrap resources
Cluster and Machine controllers should add the `cluster-name` label to external objects (i.e. bootstrap and infrastructure objects) in addition to the owner reference.

/kind feature
/milestone v0.3.0
/help
/priority important-longterm

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1567

### CODE REVIEW

File: controllers/machine_controller_phases.go [hunk 1]
```diff
@@ -86,23 +86,34 @@ func (r *MachineReconciler) reconcileExternal(ctx context.Context, m *clusterv1.
...
 
-	if !util.HasOwnerRef(obj.GetOwnerReferences(), machineOwnerRef) {
-		obj.SetOwnerReferences(util.EnsureOwnerRef(obj.GetOwnerReferences(), machineOwnerRef))
-		if err := r.Client.Patch(ctx, obj, objPatch); err != nil {
-			return nil, errors.Wrapf(err,
-				"failed to set OwnerReference on %v %q for Machine %q in namespace %q",
-				obj.GroupVersionKind(), ref.Name, m.Name, m.Namespace)
-		}
+	// Add ownerRef to object.
+	obj.SetOwnerReferences(util.EnsureOwnerRef(obj.GetOwnerReferences(), ownerRef))
```
Comment (detiber) [inline controllers/machine_controller_phases.go]: Wouldn't this potentially create an ownerRef for a Machine owned by a MachineSet? Are we intending to change that behavior for v1alpha3?

File: docs/book/src/providers/v1alpha2-to-v1alpha3.md [hunk 2]
```diff
@@ -15,11 +15,15 @@
 - The field is now required on all Cluster dependant objects.
 - The `cluster.x-k8s.io/cluster-name` label is created automatically by each respective controller.
 
-## Context is now required for `external.CloneTemplate` function.
+## Context is now required for `external.Get` and `external.CloneTemplate` function.
```
Comment (vincepri) [inline docs/book/src/providers/v1alpha2-to-v1alpha3.md]: @chuckha I'll rebase  :)

**Comment by vincepri** [comment]
@detiber Sounds like a different work item? Is there an issue already open about that? I don't even know anything about that 😂

**Comment by vincepri** [comment]
@detiber ptal

**Pull request merged**